mongo/src/test/java/com/redhat/lightblue/mongo/crud/MongoCRUDControllerTest.java

```diff
@@ -133,6 +133,23 @@ public void ensureIndexNotRecreated() throws IOException, InterruptedException {
     }
 
     @Test
+    public void findDocument_CI() throws Exception {
```

Comment:
You need a test for elemMatch. I think one way you can do this is:
 - create a doc with CI index in array element field
 - manually modify array index so array.*.field="someValue" and array.*.@mongoHidden.field="REALVALUE"
 - regex search for "realvalue", it should find it.